semeio/jobs/rft/utility.py

```diff
@@ -0,0 +1,265 @@
+
+    for line, line_number in well_time_lines:
+        tokens = line.split()
+
+        if not tokens:
+            continue
+
+        elif len(tokens) != 5:
+            err_msg = (
+                base_error_msg
+                + "Unexpected number of tokens: expected 5 got {num_tokens}"
+            )
+            raise argparse.ArgumentTypeError(
+                err_msg.format(
+                    line_number=line_number, filename=filename, num_tokens=len(tokens)
+                )
+            )
+
+        well = tokens[0]
+
+        try:
+            report_step = int(tokens[4])
+        except ValueError:
+            err_msg = base_error_msg + "Unable to convert {report_step} to int"
+            raise argparse.ArgumentTypeError(
+                err_msg.format(
+                    line_number=line_number, filename=filename, report_step=tokens[4]
+                )
+            )
+
+        try:
+            year = int(tokens[3])
+            month = int(tokens[2])
+            day = int(tokens[1])
+            time = datetime.date(year, month, day)
+
+        except ValueError:
+            err_msg = (
+                base_error_msg
+                + "Unable to parse date, expected day month year got: {day} {month} {year}"
+            )
+            raise argparse.ArgumentTypeError(
+                err_msg.format(
+                    line_number=line_number,
+                    filename=filename,
+                    day=tokens[1],
+                    month=tokens[2],
+                    year=tokens[3],
+                )
+            )
+
+        well_times.append((well, time, report_step))
+
+    return well_times
+
+
+def load_and_parse_zonemap_file(filename):
+    """
+    The file must be on the format of <INT STRING*> (any number of strings)
+    which represents <k_value zone_names..>
+
+    Parameters
+    ----------
+    filename : string
+        Filename to open
+
+    Returns
+    -------
+    Dict
+        Returns a dict with k_value as keys and a list of corresponding
+        zonenames
+    """
+
+    zones_at_k_value = {}
+
+    with open(filename, "r") as f:
+        zonemap_lines = f.readlines()
+
+    zonemap_lines = [(strip_comments(l), i + 1) for i, l in enumerate(zonemap_lines)]
+    basic_err_msg = "Line {line_number} in ZoneMap file {filename} not on proper format: 'k zonename <zonename> ...'. "
+    for line, line_number in zonemap_lines:
+        zonemap_line = line.split()
+
+        if not zonemap_line:
+            continue
+
+        if len(zonemap_line) < 2:
+            raise argparse.ArgumentTypeError(
+                basic_err_msg
+                + "Number of zonenames can be 1 or more".format(
+                    line_number=line_number, filename=filename
+                )
+            )
+
+        try:
+            raw_k = int(zonemap_line[0])
+        except ValueError:
+            raise argparse.ArgumentTypeError(
+                basic_err_msg
```

Comment:
unused arguments in format i guess?

Reply:
No? The arguments are included in the `basic_err_msg` string, initialized further up. Unused arguments would have thrown an error by the way